python/tskit/combinatorics.py

```diff
@@ -272,15 +270,55 @@ def split_polytomies(
         if tree.num_children(u) > 2:
             root = TreeNode.random_binary_tree(tree.children(u), rng)
             root.label = u
-            time = tree.time(u) - epsilon
-            stack = [(child, time) for child in root.children]
+            root_time = tree.time(u)
+            stack = [(child, root_time) for child in root.children]
             while len(stack) > 0:
-                node, time = stack.pop()
+                node, parent_time = stack.pop()
                 if node.label is None:
-                    node.label = tables.nodes.add_row(time=time)
+                    if epsilon is None:
```

Comment:
I do think you are correct to retain epsilon, even if it is not usually used, because we may wish later to insert nodes between the ones we've just created. I can't actually see why that would be needed for `split_polytomies`, but it is certainly needed when we create new nodes using path compression. In other words, having the only possibility for epsilon being the minimal representable distance might restrict how the TS could be used down the line, so allowing a tweakable parameter (even if it is not the default) seems justified.